dartx/lib/src/iterable.dart

```diff
@@ -483,6 +483,45 @@ extension IterableX<E> on Iterable<E> {
     return list;
   }
 
+  /// Returns all elements matching the given [predicate].
+  Iterable<E> filter(bool predicate(E element)) => where(predicate);
+
+  /// Returns all elements that satisfy the given [predicate].
+  Iterable<E> filterIndexed(bool predicate(E element, int index)) =>
+      whereIndexed(predicate);
```

Comment:
I thought about duplicating the implementation. But I think the dart compiler is smart enough to do detect the method forwarding and replace `filter*` method with `where*` methods.
It shouldn't result in a (tiny) performance hit when compiled ahead-of-time.

Can you confirm @sjindel-google?

Reply:
This is something I'm also interested in. I think there should be more documentation on how to improve Dart performance. I'd also like to know how inlining works and how to avoid the performance hit when using `async`. This is something I struggled with when implementing Hive.